mantidproject/mantid
# Save Scaled Background Subtraction Metadata to Subtracted Files

### Description of work

#### Summary of work
When saving out background subtracted workspaces using the recently added (#35811) scaled background subtraction work, the name of the workspace used and the scale factor it was multiplied by are now saved to the NX and CanSAS1D output files. 

Fixes: #35939


**Report to:** @smk78 


#### Further detail of work
- Since the scaled background subtraction saves out the non-subtracted workspaces as well, these files do not include the additional metadata.

### To test:
1. In your Manage User Directories window, ensure that your save location is somewhere sensible. Ideally a new empty directory as it'll make finding the files you need much easier. 
2. Open the ISIS SANS Interface
3. Load the batch and user files from the loqdemo in the Training Course Data. 
4. Switch the reduction mode to HAB in the Settings tab and run the reduction. 
5. Take note of one of the output workspaces produced.
6. Make a copy of one of the lines currently in the runs table.
7. Give it a new output name. 
8. Tick the Scaled Background Subtraction checkbox
9. In your new row, enter in the name of the workspace in your workspace column
10. Enter a decimal value in the scale factor column (0.5 is what I generally use)
11. Check that the reduction mode is still HAB
12. Change the save options to output both NXCanSAS and CanSAS1D formats
13. Run the reduction for your new line
14. In your save directory, check that the values you entered in the table for scaled background workspace and scale factor are present in the process notes section of the output bgsub xml file. 
15. Use [HDFView](https://www.hdfgroup.org/downloads/hdfview/) to do the same for the `.h5` files. 

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
